src/resolvers/user.js

```diff
@@ -179,6 +181,31 @@ module.exports = {
         throw new ApolloError('Something went wrong');
       }
 
+      return true;
+    },
+
+    /**
+     * Leave workspace
+     *
+     * @param {ResolverObj} _obj
+     * @param {Workspace.id} workspaceId - id of the workspace where the user should be removed
+     * @param {Context.user} user - current authorized user {@see ../index.js}
+     * @return {Promise<boolean>} - true if operation is successful
+     * @returns {Promise<boolean>}
+     */
+    async leaveWorkspace(_obj, { workspaceId }, { user }) {
+      const team = new Team(workspaceId);
+      const users = await team.getAllUsers();
+      const member = users.find(el => el._id.toString() === user.id);
+
+      if (!member) {
+        throw new ApolloError('You are not in the workspace');
+      }
+
+      const membership = new Membership(user.id);
+      await team.removeMember(user.id);
```

Comment:
Need to research how to use MongoDB transactions in here. If the first query succeeds while the second fails it could cause a potential error.